### ISSUE TITLE
Setup AWS S3 data validation reporting: Allure

### ISSUE THREAD

**Issue by daniel-ji** [chanzuckerberg/cryoet-data-portal#999]
To be done after #996 and all other pytest s3 data validation integration / improvements are done. 

### Migration / Adaptation of Utz's work
- [x] Port over & adapt pytest-allure testing from cryoet-portal
- [x] Port over & adapt run_tests.py wrapper script from cryoet-portal

Improvement
- [ ] Integrate Allure reporting with AWS workflows. (won't do in this issue)
- [x] Continue to build Allure reporting to ensure filtering & hierarchical view is as desired (don't know how big of a task this yet is)
